tests/test_compile_guard.py

```diff
@@ -253,20 +254,50 @@ class _Top(m.Circuit):
 
 
 def test_compile_guard_select_basic():
+
     class _Top(m.Circuit):
         io = m.IO(I=m.In(m.Bit), O=m.Out(m.Bit)) + m.ClockIO()
-
         x = m.Register(m.Bit)()(io.I ^ 1)
         y = m.Register(m.Bit)()(io.I)
-
-        io.O @= m.compile_guard_select(
-            COND1=x, COND2=y, default=io.I
-        )
+        io.O @= m.compile_guard_select(COND1=x, COND2=y, default=io.I)
 
     basename = "test_compile_guard_select_basic"
     m.compile(f"build/{basename}", _Top, inline=True)
     assert m.testing.check_files_equal(
-        __file__, f"build/{basename}.v", f"gold/{basename}.v")
+        __file__, f"build/{basename}.v", f"gold/{basename}.v"
+    )
+
+
+def test_compile_guard_select_two_keys():
+
+    class _Top(m.Circuit):
+        io = m.IO(I=m.In(m.Bit), O=m.Out(m.Bit)) + m.ClockIO()
+        x = m.Register(m.Bit)()(io.I ^ 1)
+        io.O @= m.compile_guard_select(COND1=x, default=io.I)
+
+    basename = "test_compile_guard_select_two_keys"
+    m.compile(f"build/{basename}", _Top, inline=True)
+    assert m.testing.check_files_equal(
+        __file__, f"build/{basename}.v", f"gold/{basename}.v"
+    )
```

Comment:
Should we update these to use mlir/mlir-verilog instead of coreir? Will save us doing it in the future.

Reply:
Will do these in a follow up PR.